bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,437 @@
+  description?: string;
+}
+
+interface ApiHistory {
+  added?: ChangeSchema[];
+  deprecated?: ChangeSchema[];
+  changes?: ChangeSchema[];
+}
+
+interface Options {
+  checkPlacement: boolean;
+  // TODO: Implement this when GH_TOKEN isn't needed to fetch PR release versions anymore
+  // checkPullRequestLinks: boolean;
+  breakingChangesFile: string;
+  checkStrings: boolean;
+  ignoreGlobs: string[];
+  schema: string;
+}
+
+interface HTMLWithPreviousNode extends HTML {
+  previousNode?: Node;
+}
+
+export async function findPossibleApiHistoryBlocks(
+  content: string,
+): Promise<HTMLWithPreviousNode[]> {
+  const { fromMarkdown } = (await dynamicImport('mdast-util-from-markdown')) as {
+    fromMarkdown: typeof FromMarkdownFunction;
+  };
+  const { visit } = (await dynamicImport('unist-util-visit')) as {
+    visit: typeof VisitFunction;
+  };
+  const tree = fromMarkdown(content);
+  const codeBlocks: HTMLWithPreviousNode[] = [];
+
+  visit(
+    tree,
+    // Very loose check for YAML history blocks to help catch user error
+    (node) =>
+      node.type === 'html' &&
+      (node as Literal<string>).value.includes('```') &&
+      (node as Literal<string>).value.toLowerCase().includes('yaml') &&
+      (node as Literal<string>).value.toLowerCase().includes('history'),
+    (node: Node, index) => {
+      // We don't want to modify the original node
+      // ? Maybe just copy the previous node's type instead of the whole node
+      const shallowNodeCopy = { ...node } as HTMLWithPreviousNode;
+      if (index !== null) {
+        const previousNode = tree.children[index - 1];
+        if (previousNode) shallowNodeCopy.previousNode = previousNode;
+      }
+      codeBlocks.push(shallowNodeCopy);
+    },
+  );
+
+  return codeBlocks;
+}
+
+type LintingResults = {
+  historyBlockCounter: number;
+  documentCounter: number;
+  errorCounter: number;
+  warningCounter: number;
+};
+
+async function main(
+  workspaceRoot: string,
+  globs: string[],
+  { checkPlacement, breakingChangesFile, checkStrings, schema, ignoreGlobs = [] }: Options,
+): Promise<LintingResults> {
+  let documentCounter = 0;
+  let historyBlockCounter = 0;
+  let errorCounter = 0;
+  let warningCounter = 0;
+
+  try {
+    const { fromHtml } = (await dynamicImport('hast-util-from-html')) as {
+      fromHtml: typeof FromHtmlFunction;
+    };
+    const { fromMarkdown } = (await dynamicImport('mdast-util-from-markdown')) as {
+      fromMarkdown: typeof FromMarkdownFunction;
+    };
+
+    const workspace = new DocsWorkspace(workspaceRoot, globs, ignoreGlobs);
+
+    let validateAgainstSchema: ValidateFunction<ApiHistory> | null = null;
+
+    if (schema) {
+      try {
+        const ajv = new Ajv();
+        const ApiHistorySchemaFile = await readFile(schema, { encoding: 'utf-8' });
+        const ApiHistorySchema = JSON.parse(ApiHistorySchemaFile) as JSONSchemaType<ApiHistory>;
+        validateAgainstSchema = ajv.compile(ApiHistorySchema);
```

Comment:
```suggestion
        const ApiHistorySchema = JSON.parse(ApiHistorySchemaFile);
        validateAgainstSchema = ajv.compile<ApiHistory>(ApiHistorySchema);
```

Avoid the type assertion - in this case there's no functional, it's just good practice to avoid type assertions unless they're the only option.